internal/mode/static/telemetry/collector.go

```diff
@@ -275,3 +273,46 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+type clusterInformation struct {
+	Platform  string
+	Version   string
+	ClusterID string
+}
+
+func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (clusterInformation, error) {
+	var clusterInfo clusterInformation
+
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+	if len(nodes.Items) == 0 {
+		return clusterInformation{}, errors.New("failed to collect cluster information: NodeList length is zero")
+	}
+	node := nodes.Items[0]
+
+	var version *k8sversion.Version
+	clusterInfo.Version = "unknown"
+	kubeletVersion := node.Status.NodeInfo.KubeletVersion
```

Comment:
Is the Kubelet Version guaranteed to be set?

Reply:
No, which my thought was then kubeletVersion would be set to "", and would cause an error in `ParseGeneric` thus clusterInfo.Version would be set to "unknown" . 

Is this of concern?

Reply:
I see. That work, ty for the explanation!